### ISSUE TITLE
Please consider that making BucketLookupDNS as default value of BucketLookupType when initialize a client

### ISSUE THREAD

**Issue by ChaosBee** [minio/minio#16764]
**Is your feature request related to a problem? Please describe.**

When add new s3-compatible tier configuration, which doesn't support path style access, you will get Error. 
As Metioned by S3--_Currently, Amazon S3 supports both virtual-hosted–style and path-style URL access in all AWS Regions. However, path-style URLs will be discontinued in the future._  We know that virtual-hosted–style will be the actual standard of s3-compatible service.

Reference: https://docs.aws.amazon.com/AmazonS3/latest/userguide/RESTAPI.html

**Describe the solution you'd like**
Make BucketLookupDNS as default value of BucketLookupType or provide an entry point of setting BucketLookupType.

**Describe alternatives you've considered**
Push minio-go sdk maintainer to change behaviour that only change auto to dns only for Amazon/Google  storage. 
[
_default to virtual only for Amazon/Google  storage_](https://github.com/minio/minio-go/blob/master/api.go#L961)

Thank you for your efforts.

**Comment by harshavardhana** [comment]
Virtual Bucket DNS is automatic based on Hostname regex. If you want it to be default as "on" then you should set that in your minioClient constructor.

**Comment by ChaosBee** [comment]
@harshavardhana   See you again. I could understand minio-go's behaviour when handling BucketLookupAuto. But as _Virtual Bucket DNS_ is encouraged by S3 and S3 is the actual standard of object storage. So I think that maybe the auto logic should prefer _Virtual Bucket DNS_. Just an advice.

I opened up a issue here, cause I found that most of minioClient initilizations in minio service would not try to set BucketLookupType and did not provide an entry setting it. 


Thanks again for your efforts.
